src/SmartComponents/DriftPage/DriftTable/DriftTable.js

```diff
@@ -183,11 +216,23 @@ export class DriftTable extends Component {
                 return sys.id === item.id;
             });
 
-            row.push(
-                <td className={ fact.state === 'DIFFERENT' ? 'highlight comparison-cell' : 'comparison-cell' }>
-                    { system.value === null ? 'No Data' : system.value }
-                </td>
-            );
+            if (this.referenceId) {
+                row.push(
+                    <td className={
+                        system.meets_reference_expectations || system.meets_reference_expectations === undefined
+                            ? 'comparison-cell'
+                            : 'highlight comparison-cell different-fact-cell' }
+                    >
+                        { system.value === null ? 'No Data' : system.value }
+                    </td>
+                );
+            } else {
+                row.push(
+                    <td className={ fact.state === 'DIFFERENT' ? 'highlight comparison-cell' : 'comparison-cell' }>
```

Comment:
Looks like the only difference between the two `row.push` statements is how this className is set. I think then you can extract that to logic thats wrapped in an if-else and call row.push only once.